### PR TITLE
Add throttling classes for thumbnail endpoint

### LINKED ISSUE #863: Adjust throttling of thumbnail endpoints
## Problem

Currently thumbnail requests are throttled at a rate of one thousand per minute:

https://github.com/WordPress/openverse-api/blob/e9cfd3cc705c573e3f6ab2b632737e94bb82aee0/api/catalog/api/views/image_views.py#L96

When used by unauthenticated users making requests rapidly, as is common in scraping workflows, this can potentially send a lot of traffic to our API partners which may lead to rate limiting or denylisting of our API from their endpoints.

## Description

We should reduce the unauthenticated thumbnail throttling to something like `150/min`. This seems a reasonable amount for anyone building a user interface that displays thumbnails on-demand, which is more in-line with our intended use cases for the API. 

The throttling for _authenticated_ accounts should be higher, perhaps `500/min`?

## Additional context

We have also discussed implementing CSRF tokens on the front-end to solve this problem, which would be used in conjunction with this approach but with the unauthenticated throttling even more aggressive.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by krysal**
## Fixes

Fixes #863 by @zackkrida

## Description

As title says, adds the `AnonThumbnailRateThrottle` and `OAuth2IdThumbnailRateThrottle` classes.

## Testing Instructions

1. Set the following variables in your `end.docker` file
```
DISABLE_GLOBAL_THROTTLING=False
THROTTLE_ANON_THUMBS=2/minute
# You may also need to add "localhost" to the ALLOWED_HOSTS
```
2. `just up`
3. Try viewing 3 thumbnails (or just one refreshing more than twice)
4. You should get throttled in this endpoint
5. Repeat for the authenticated (**verified**) version 

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [N/A] I added or updated tests for the changes I made (if applicable).
- [N/A] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: api/catalog/api/utils/throttle.py [hunk 1]
```diff
@@ -61,6 +61,10 @@ class OneThousandPerMinute(AbstractAnonRateThrottle):
     rate = "1000/min"
 
 
+class OneHundredFiftyPerMinute(AbstractAnonRateThrottle):
+    rate = "150/minute"
```
Comment (sarayourfriend) [inline api/catalog/api/utils/throttle.py:65]: It _mayyyy_ be a good idea (given we're kind of guessing at a good rate limit for this) to make this an environment variable and change the class have a specific name for thumbs:

```suggestion
class AnonThumbnailThrottle(AbstractAnonRateThrottle):
    rate = config("ANON_THUMBNAIL_THROTTLE_RATE", "150/minute")
```

What do you think? This way if we find that it needs to be adjusted we don't have to make a code change, cut a new release, and then redeploy, it'd just be a redeploy.

Comment (sarayourfriend) [reply api/catalog/api/utils/throttle.py:65]: Okay, I think I have my final suggested change here to make it configurable via the environment:

1. Rename the throttle to AnonThumbnailThrottle and add a scope
2. Add the scope to the `DEFAULT_THROTTLE_RATES` in `settings.py` and set the value to an environment variable similar to how the anon throttle rates are set

It's a minor change in the context of this PR but it puts the configuration added here more in line with the rest of our throttles.

Comment (krysal) [reply api/catalog/api/utils/throttle.py]: I applied the suggested changes but doesn't seem to work well for authenticated requests, so I will convert it to draft.

**Comment by krysal** [comment]
Not sure what I was doing before but I tested it again and it works, lol. The existing unit tests seem to cover the newly classes added, as they're part of `AbstractOAuth2IdRateThrottle` & `AbstractAnonRateThrottle` subclasses, robust tests! :)

**Review by dhruvkb — APPROVED** [review]
Good change, should we remove the very numeric throttle classes (`OneThousandPerMinute`/`OnePerSecond` etc.) now that they're unused?

**Comment by krysal** [comment]
@dhruvkb Fair point, I agree on removing them in another PR for clean-up.

**Pull request merged**